gtsam/hybrid/HybridGaussianFactor.h

```diff
@@ -80,26 +85,26 @@ class GTSAM_EXPORT HybridGaussianFactor : public HybridFactor {
    * @param continuousKeys A vector of keys representing continuous variables.
    * @param discreteKeys A vector of keys representing discrete variables and
    * their cardinalities.
-   * @param factors The decision tree of Gaussian factors stored
-   * as the mixture density.
+   * @param factors The decision tree of Gaussian factors and arbitrary scalars.
    */
   HybridGaussianFactor(const KeyVector &continuousKeys,
                        const DiscreteKeys &discreteKeys,
-                       const Factors &factors);
+                       const FactorValuePairs &factors);
 
   /**
    * @brief Construct a new HybridGaussianFactor object using a vector of
    * GaussianFactor shared pointers.
    *
    * @param continuousKeys Vector of keys for continuous factors.
    * @param discreteKeys Vector of discrete keys.
-   * @param factors Vector of gaussian factor shared pointers.
+   * @param factors Vector of gaussian factor shared pointers
```

Comment:
Can we kill this constructor? Error-prone. Simple change.

Reply:
`DecisionTree` has a vector based constructor and this follows that. Should we leave the correctness to the user? The assumption here is that they understand how the `DecisionTree` class creates the tree though.

Reply:
I prefer to kill it. Where is it currently used ?

Reply:
The `DecisionTree` constructor from vectors? I believe in a bunch of places for discrete stuff.

Reply:
Updated/Removed all vector constructors in the Hybrid API. :)
HybridNonlinearFactor and HybridGaussianFactor only accept a single discrete key, while HybridGaussianConditional doesn't have a vector constructor at all.